Use Ajax call to load organizations on index page

cyhsutw opened the pull request:
Hello guys,

Just implemented to use AJAX calls to load organizations on index page.

Fixes #504.

Demo:

1. For new users without any classrooms available:

![none](https://cloud.githubusercontent.com/assets/1046917/14233406/270d9dde-f9fa-11e5-9500-e7dcf264e8bb.gif)

2. For returning users with some new classrooms (there's a tiny Octocat spinner on the right of the `Classrooms`):

![some](https://cloud.githubusercontent.com/assets/1046917/14233411/419978bc-f9fa-11e5-9d92-d4bf3251ab06.gif)

It will be appreciated if you could give me some feedback/suggestion. :smiley: 

Thanks!

tarebyte commented on config/routes.rb:
```diff
@@ -44,6 +44,8 @@
     end
   end
 
+  get '/classrooms-async', to: 'organizations#async_index'
```
This should be scoped underneath line 34

cyhsutw replied:
Thanks! I'll fix this with my next commit. :smile:

tarebyte commented on app/controllers/organizations_controller.rb:
```diff
@@ -1,17 +1,22 @@
-    @organizations = current_user.organizations.includes(:users).page(params[:page])
+  end
+
+  def async_index
```
I'm not a fan of the name at all tbh.

@johndbritton do you have any other suggestions of what we could call this?

cyhsutw replied:
Actually I was using `list` but I found it's a little bit confusing (or maybe it's not). 

I would love to hear suggestions from you!

tarebyte replied:
@cyhsutw what about `users_organizations`

johndbritton replied:
I haven't used this kind of pattern before, so I'm sorry I'm not much help with the naming here. Are there any examples that we can look at elsewhere?

tarebyte replied:
So this kind of matches this endpoint

https://github.com/education/classroom/blob/master/app/controllers/stafftools/resources_controller.rb#L8-L12

johndbritton commented:
@tarebyte I'll defer to you on if you think this is the right approach.

johndbritton commented on app/controllers/organizations_controller.rb:
```diff
@@ -1,17 +1,22 @@
+# rubocop:disable ClassLength
```
Maybe it's time to start breaking some of this logic out into `lib` with service objects.

cyhsutw commented:
Hi @tarebyte @johndbritton 

Thanks for reviewing the code.

I've renamed the routes and method to `user_organization` and the endpoint is now at `/classrooms/user-organizations`.

Also, I extracted some of the code to `service/organization_service.rb`, not sure if I'm doing it right.

I would love to hear feedback/suggestion from you.

nwoodthorpe commented:
This PR has been stale for over a year now.
